packages/gen-ai/frontend/src/app/services/llamaStackService.ts

```diff
@@ -156,3 +168,50 @@ export const getLSDstatus = (project: string): Promise<LlamaStackDistributionMod
       );
     });
 };
+
+/**
+ * Fetches all available AI models from the AI Assets API
+ * @param namespace - The namespace to fetch models for
+ * @returns Promise<AIModel[]> - Array of available AI models with their metadata
+ * @throws Error - When the API request fails or returns an error response
+ */
+export const getAIModels = (namespace: string): Promise<AIModel[]> => {
+  const url = `${URL_PREFIX}/api/v1/aa/models?namespace=${namespace}`;
+  return axios
+    .get(url)
+    .then((response) => {
+      // Transform the API response to match our AIModel interface
+      const data = response.data.data || response.data;
+      return data.map((item: AAModelResponse, index: number) => {
+        // Parse endpoints to extract internal and external endpoints
+        const internalEndpoint =
+          item.endpoints.find((ep) => ep.startsWith('internal:'))?.replace('internal: ', '') ||
+          null;
+        const externalEndpoint =
+          item.endpoints.find((ep) => ep.startsWith('external:'))?.replace('external: ', '') ||
+          null;
+
+        return {
+          id: `model-${index}`,
+          name: item.model_name,
+          description: item.description || 'No description available',
+          internalEndpoint,
+          externalEndpoint,
+          useCase: item.usecase || 'General purpose',
+          playgroundStatus: 'available', // All models are available for playground
```

Comment:
I thought this would be whether the model itself (the `InferenceService` CR) is available. If not, we cannot use it. You can check `Model deployment` page to see some similar fields.